Linux networking support for HBS

Goosebirb opened the pull request:
Replicates or expands upon functionality in the windows version, inc.
Wifi Test
NAT test (expanded to include CG-NAT checks)
NIC driver checks
Cloudflare speed test
System clock sync

helldivers2fixes commented:
Appreciate the contribution! Can you re-write with only the changed functions modified? Otherwise, I can close this pull request, and incorporate the features at some point in the future

Goosebirb commented:
I removed the additional experimental checks, and the CG-NAT test portion is commented out, allowing for functionality to be easily reimplemented.
I'm not sure if I updated my code properly, but hopefully it shows as changed in the PR. I fnot, nuke the PR and I will resubmit a fresh one
Also implemented checks for whether the OS is running linux to trigger the related linux network checks
